chart/templates/deployment-sidekiq.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
     {{- include "mastodon.labels" . | nindent 4 }}
 spec:
   {{- if not .Values.autoscaling.enabled }}
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ .Values.mastodon.sidekiq.replicaCount | default .Values.replicaCount }}
```

Comment:
Fallback on `.Values.replicaCount` won't work since you've already defined a default value for `.Values.mastodon.sidekiq.replicaCount`
See : 

```YAML
# test.values.yaml
replicaCount: 5
mastodon:
  secrets:
    secret_key_base: "abc"
    otp_secret: "def"
    vapid:
      private_key: "ghi"
      public_key: "klm"
```

```
helm template -f test.values.yaml . -s templates/deployment-web.yaml | grep replicas
  replicas: 1
```

Best way imo is to just ditch the `.Values.replicaCount` and bump a major with Upgrading steps in the readme

Reply:
Ah damn, you're right >_<. I'm happy to bump the chart's major version, or simply comment out the deployment specific replica counts to keep things less painful.

Do you have a preference?

Reply:
imo it's safer to remove the global replicaCount, this could lead to unwanted issues like having 5 sidekiq schedulers

Reply:
imo it's safer to remove the default / global replicaCount, this could lead to issues like having 5 sidekiq schedulers